lib/galaxy/webapps/galaxy/controllers/admin.py

```diff
@@ -1402,10 +1403,22 @@ def reset_user_password(self, trans, payload=None, **kwd):
             return self.message_exception(trans, 'Please specify user ids.')
 
     def _delete_user(self, trans, ids):
+        gdpr_compliant = self.app.config.gdpr_compliance_mode
+
         message = 'Deleted %d users: ' % len(ids)
         for user_id in ids:
             user = get_user(trans, user_id)
             user.deleted = True
+
+            if gdpr_compliant:
+                # Replace email + username with a (theoretically) unreversable
+                # hash. If provided with the username we can probably re-hash
+                # to identify if it is needed for some reason.
+                #
+                # Deleting multiple times will re-hash the username/email
+                user.email = new_secure_hash(user.email)
```

Comment:
this user can now register again, what will happen when they try to delete themselves again?

Reply:
... eww. yeah, ok. I guess I'll throw a `random.random()` in there.